src/app.ts

```diff
@@ -17,7 +17,11 @@ export function createApp(
     const config = createProxyConfig(options);
     const { logger } = config;
     logger.debug('Configuration:', config);
-    const client = unleashClient || createClient(config);
+    const client =
+        unleashClient ||
+        (options.clientMode === 'new'
+            ? createNewClient(config)
+            : createSingletonClient(config));
```

Comment:
Should we have this as an option or just another method `createNonSingletonApp`? I'd like it to be `createSingletonApp` but that basically requires changing the current method behavior.

I'm ok with this approach I'm just wondering if you considered having a new method instead

Reply:
I though that for our own code using proxy it'd be easiest to do:  unleashProxy.createApp({clientMode: 'new'}). Test it in production to find if there are no corner cases. Then once we're happy to remove clientMode config and switch to createNewClient for everyone and rename it back to createClient.